src/main/java/com/google/devtools/build/lib/cmdline/PackageIdentifier.java

```diff
@@ -116,7 +119,9 @@ public static PackageIdentifier discoverFromExecPath(PathFragment execPath, bool
   private PackageIdentifier(RepositoryName repository, PathFragment pkgName) {
     this.repository = Preconditions.checkNotNull(repository);
     this.pkgName = Preconditions.checkNotNull(pkgName);
-    this.hashCode = Objects.hash(repository, pkgName);
+    this.repositoryCasing = repository.getName();
```

Comment:
I don't see why you need to make a copy here. The RepositoryName and pkgName aren't changing underneath you, are they?

Reply:
RepositoryName.equals is case-ignoring. If I only stored the RepositoryName reference, I couldn't tell apart PackageIdentifiers with differently cased RepositoryNames. So I need to also store the String representation.